src/mlpack/methods/grad_boosting/grad_boosting.hpp

```diff
@@ -0,0 +1,217 @@
+/**
+ * @file methods/grad_boosting/grad_boosting.hpp
+ * @author Abhimanyu Dayal
+ *
+ * Gradient Boosting class. Gradient Boosting uses weak learners (primarily 
+ * decision stumps), and trains them sequentially, such that future learners are 
+ * trained to detect the errors (or gradients) of previous learners. 
+ * The results obtained from all of these learners are subsequently 
+ * aggregated to give a final result.
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+// Include guard - Used to prevent including double copies of the same code
+#ifndef MLPACK_METHODS_GRAD_BOOSTING_GRADBOOSTING_HPP
+#define MLPACK_METHODS_GRAD_BOOSTING_GRADBOOSTING_HPP
+
+// Importing base components required to write mlpack methods.
+#include <mlpack/core.hpp>
+
+// Only using decision trees for now, therefore only including decision tree functionalities.
+#include <mlpack/methods/decision_tree/decision_tree_regressor.hpp>
+
+
+// Written in mlpack namespace.
+namespace mlpack {
+/**
+ * The Gradient Boosting class. Gradient Boosting is a boosting algorithm, meaning that it
+ * combines an ensemble of weak learners to produce a strong learner.
+ * 
+ * Gradient Boosting is generally implemented using Decision Trees, or more specifically 
+ * Decision Stumps i.e. weak learner Decision Trees with low depth.
+ * 
+ * @tparam MatType Data matrix type (i.e. arma::mat or arma::sp_mat).
+ */
+
+
+template<typename MatType = arma::mat>
+class GradBoosting 
+{
+ public: 
+
+  // Defined DecisionTree with template as WeakLearnerType to avoid repetition
```

Comment:
```suggestion
  // Convenience typedef for the type of weak learner we are using.
```
I think this is slightly clearer, let me know what you think.